src/helpers.c

```diff
@@ -461,6 +461,50 @@ void remove_trailing_slash(char *url)
 	}
 }
 
+static bool adjust_system_time()
+{
+	int err;
+	time_t clear_time, system_time;
+	char *time_str;
+
+	err = verify_time(globals.path_prefix, &system_time, &clear_time);
+	if (err == -ENOENT) {
+		/* in the case we are doing an installation to an empty directory
+	/	 * using swupd verify --install, we won't have a valid versionstamp
+		 * in path_prefix, so try searching without the path_prefix */
+		err = verify_time(NULL, &system_time, &clear_time);
+	}
+
+	if (err == 0) {
+		// System time is correct
+		return true;
+	}
+
+	if (err == -ENOENT) {
+		error("Failed to read system timestamp file\n");
+	}
+
+	time_str = time_to_string(system_time);
+	if (time_str) {
+		printf("Current time is %s\nAttempting to fix...\n", time_str);
+		FREE(time_str);
+	}
+
+	if (set_time(clear_time)) {
+		time_str = time_to_string(clear_time);
+		if (time_str) {
+			info("Set system time to %s\n", time_str);
+			FREE(time_str);
+		}
+
+		// System time has been fixed
+		return true;
+	}
+
+	// System time is wrong
```

Comment:
Shouldn't we warn about not being able to fix the time? After all the last message the user would have seen was "Attempting to fix it..."